services/httpd/handler.go

```diff
@@ -615,21 +618,53 @@ func resultError(w http.ResponseWriter, result influxql.Result, code int) {
 
 // Filters and filter helpers
 
-// parseCredentials returns the username and password encoded in
-// a request. The credentials may be present as URL query params, or as
-// a Basic Authentication header.
-// as params: http://127.0.0.1/query?u=username&p=password
-// as basic auth: http://username:password@127.0.0.1
-func parseCredentials(r *http.Request) (string, string, error) {
+type credentials struct {
+	Method   AuthenticationMethod
+	Username string
+	Password string
+	Token    string
+}
+
+// parseCredentials parses a request and returns the authentication credentials.
+// The credentials may be present as URL query params, or as a Basic
+// Authentication header.
+// As params: http://127.0.0.1/query?u=username&p=password
+// As basic auth: http://username:password@127.0.0.1
+// As Bearer token in Authorization header: Bearer <JWT_TOKEN_BLOB>
+func parseCredentials(r *http.Request) (*credentials, error) {
 	q := r.URL.Query()
 
-	if u, p := q.Get("u"), q.Get("p"); u != "" && p != "" {
-		return u, p, nil
+	// Check for the HTTP Authorization header.
+	if s := r.Header.Get("Authorization"); s != "" {
```

Comment:
Never mind, I didn't realize `Authorization` was a standard header these days. Typically when using custom headers, you always prepend them with `X-` first.